lib/DbUrlList.js

```diff
@@ -179,7 +179,7 @@ DbUrlList.prototype._calcNextRetryDate = function (numErrors) {
     // If we want to schedule a crawl now, we subtract a random number of
     // seconds. This ensures the order we crawl URLs is random; otherwise, if
     // we parse a sitemap, we could get stuck crawling one host for hours.
-    delay = - Math.random() * YEAR_MS;
+    delay = - Math.random() * this._recrawlInMs;
```

Comment:
I don't think this has the intended effect. Notice that `delay` is negative here. This is simply to randomize new URLs that come onto the queue.